cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -1966,4 +1966,233 @@ const char* binary_string(gdv_int64 context, const char* text, gdv_int32 text_le
   *out_len = j;
   return ret;
 }
+
+FORCE_INLINE
+const char* parse_url_utf8_utf8(gdv_int64 context, const char* url, gdv_int64 url_len,
+                                const char* part_to_extract, gdv_int64 part_len,
+                                gdv_int32* out_len) {
+  if (url_len == 0) {
+    return nullptr;
+  }
+
+  auto protocol_key = "PROTOCOL";
```

Comment:
Maybe we can check a way of not having to instantiate these variables constantly.